### ISSUE TITLE
summary should be interactive

### ISSUE THREAD

**Issue by sh0ji** [jsx-eslint/eslint-plugin-jsx-a11y#656]
I believe the `summary` element should be considered an interactive element and that there shouldn't be any errors thrown when a `click` listener is attached to it:

```jsx
// currently flags the following:
// jsx-a11y/click-events-have-key-events
// jsx-a11y/no-static-element-interactions
<summary onClick={clickHandler} />
```

[The spec does list `details` element as interactive](https://html.spec.whatwg.org/multipage/dom.html#interactive-content), but this is misleading since [its activation behavior](https://html.spec.whatwg.org/multipage/interactive-elements.html#the-details-element:activation-behaviour) is triggered by its child `summary` OR a UA-provided affordance (all browsers add a shadow `summary` to accomplish this affordance). The spec for `summary` does also describe [the `summary` activation behavior](https://html.spec.whatwg.org/multipage/interactive-elements.html#the-summary-element:activation-behaviour), making it more clear that the `summary` is the element that activates the `details` open state.

This all also makes logical sense since the `details` can _contain_ interactive content, meaning it cannot itself be interactive:

```html
<details>
  <summary>More Information</summary>
  <!-- this link couldn't be interactive if the parent details were interactive -->
  <p>The <a href="#">details element</a> is fun!</p>
</details>
```

Further, both browsers and screen readers treat the `summary` like a `button` element (some screen readers actually call it an expandable button), including the same keyboard interaction model as a `button`: it is focusable without `tabindex` and <kbd>Enter</kbd> triggers `click` repeatedly when held (i.e., on `keydown`); <kbd>Space</kbd> triggers `click` on `keyup` only. I made a simple CodePen to demo/test this: https://codepen.io/sh0ji/pen/OJPppER.

Note that this also means that #582 is not a false positive.

**Comment by jessebeach** [comment]
@sh0ji you're totally right, `summary` should be interactive. Let me figure out where to fix this. It's probably in aria-query or axobject-query. Great sleuthing here!

https://w3c.github.io/html-aria/#el-summary
https://w3c.github.io/html-aria/#el-details

**Comment by jessebeach** [comment]
This was closed erroneously. This issue is still present.

**Comment by stuartjnelson** [comment]
I know this is bad open-source etiquette. We're also experiencing this problem currently. Does anyone have any pointers about what we'd need to do in order to fix? I'd be happy putting together a PR

**Comment by jessebeach** [comment]
@stuartjnelson hello! Yes, a PR would be very helpful. I start my holiday PTO in a few days and I'll have time to focus on this project next week, so your timing is perfect.
